com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollUploadAppStatusExecution.java

```diff
@@ -15,32 +15,17 @@
 public class PollUploadAppStatusExecution implements AsyncExecution {
 
     @Override
-    public AsyncExecutionState execute(ExecutionWrapper execution) throws SLException {
-        // Get the next cloud application from the context
-        final CloudApplication app = StepsUtil.getApp(execution.getContext());
+    public AsyncExecutionState execute(ExecutionWrapper execution) {
+        CloudApplication app = StepsUtil.getApp(execution.getContext());
 
         try {
             execution.getStepLogger()
                 .debug(Messages.CHECKING_UPLOAD_APP_STATUS, app.getName());
-            String status = (String) execution.getContext().getVariable(Constants.VAR_UPLOAD_STATE);
-            if (AsyncExecutionState.ERROR.name()
```

Comment:
I suppose this was the case "This should not happen" :D

Reply:
It could've happened if the previous step was using synchronous upload and it failed, but since synchronous upload is no longer used, yes this should not happen. :D